### PR TITLE
FIX: Avoid redirection when not leaving forum.

### PR DESCRIPTION
**Pull request by nbianca**
Please review this one carefully, the code is pretty old & messy.

### CODE REVIEW

File: app/assets/javascripts/discourse/lib/click-track.js.es6 [hunk 1]
```diff
@@ -128,16 +128,18 @@ export default {
...
-        data: {
-          url: href,
-          post_id: postId,
-          topic_id: topicId,
-          redirect: false
-        },
-        dataType: "html"
-      });
+    if (isInternal && !$link.hasClass("attachment")) {
+      if (tracking) {
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/lib/click-track.js.es6]: 👍 but we definitely need a test here.

File: app/assets/javascripts/discourse/lib/url.js.es6 [hunk 2]
```diff
@@ -187,13 +187,7 @@ const DiscourseURL = Ember.Object.extend({
...
     const baseUri = Discourse.BaseUri;
 
-    // If we have a baseUri and an absolute URL, make sure the baseUri
-    // is the same. Otherwise we could be switching forums.
-    if (
-      baseUri &&
-      path.indexOf("http") === 0 &&
-      pathname.indexOf(baseUri) !== 0
-    ) {
+    if (!DiscourseURL.isInternal(path)) {
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/lib/url.js.es6]: Are there tests ensuring this is the same behavior?

Comment (nbianca) [reply app/assets/javascripts/discourse/lib/url.js.es6]: Hmm... I think this might behave different in subdirectory installations.

Comment (ZogStriP) [reply app/assets/javascripts/discourse/lib/url.js.es6]: Yeah, you definitely don't want to break subfolder support ;)

Comment (nbianca) [reply app/assets/javascripts/discourse/lib/url.js.es6]: I tested this on a subfolder setup and looks good to me.

**Pull request merged**